server/resources/resource_management.go

```diff
@@ -435,15 +436,25 @@ func CreateOrUpdateResource(
 		tenantIDs := policy.GetTenantIDFilter(schema.ActionUpdate, auth.TenantID())
 		filter := transaction.IDFilter(resourceID)
 
+		_, fetchErr := preTransaction.Fetch(resourceSchema, filter, nil)
+		if fetchErr != nil {
+			exists = false
+			return nil
+		}
+		exists = true
+
 		if tenantIDs != nil {
 			filter["tenant_id"] = tenantIDs
 		}
+		policy.AddCustomFilters(filter, auth.TenantID())
 
-		_, fetchErr := preTransaction.Fetch(resourceSchema, filter, nil)
-		exists = fetchErr == nil
+		_, fetchErr = preTransaction.Fetch(resourceSchema, filter, nil)
```

Comment:
inefficiency: is it possible to stay with 1 SQL query instead of 2?

Reply:
true - previous version with one select was slightly incorrect as discussed with previous comments - when one tenant created resource with ID "A" and the second tries to create resource with same ID using PUT it would get DB error that ID already exists in DB.
We can have just SELECT w/o WHERE clause at all and then manually check if tenant ID is correct - however, it will require a lot of changes to do the same with other properties defined with "OR" or "AND".

I can instead swap this select statements - in a typical case when tenant updates own resource there would be just one select. Only when a tenant would try to create resource with PUT or would try to update resource which does not belong to him there would be two selects.